src/index.tsx

```diff
@@ -157,9 +148,42 @@ var saveToWebmarkFolder = (url: string | undefined, title: string | undefined):
     );
 }
 
-var getRandomUrlFromFolder = (): string => {
-    //TODO: implement getRandomUrlFromFolder
-    return 'https://example.com/random';
+let getRandomUrlFromFolder = (): void => {
+    chrome.storage.sync.get(
+        ['webmarkFolderId'],
+        (result?) =>{
+            console.log('webmarkFolderId found.');
+            let webmarkFolderId: string = result!['webmarkFolderId'];
+            chrome.bookmarks.getSubTree(
+                webmarkFolderId,
+                (bookmarkTreeNodes) =>{
+                    let urlList : Array<string> = [];
+                    for (let node of bookmarkTreeNodes) {
+                        if(node.children != undefined){
+                            for(let child of node.children){
+                                if(child.url!=undefined){
+                                    urlList.push(child.url);
+                                }
+                            }
+                        }
+                    }
+                    let randomUrl : string= urlList[Math.floor(Math.random() * urlList.length)];
```

Comment:
👍 
On second thought, I think it's a good practice to make the index a separate variable.